index.js

```diff
@@ -66,45 +67,85 @@ class Assets {
     }
   }
 
-  deployS3() {
-    let assetSets = this.config.targets;
-
-    // glob
-    return new Promise(resolve => {
-      assetSets.forEach(assets => {
-        const bucket = assets.bucket;
-        const prefix = assets.prefix || '';
-        assets.files.forEach(opt => {
-          this.log(`Bucket: ${bucket}:${prefix}`);
-
-          if(this.options.bucket && this.options.bucket !== bucket) {
-            this.log('Skipping');
-            return;
-          }
-
-          this.log(`Path: ${opt.source}`);
-
-          const cfg = Object.assign({}, globOpts, {cwd: opt.source});
-          glob.sync(opt.globs, cfg).forEach(filename => {
-
-            const body = fs.readFileSync(path.join(opt.source, filename));
-            const type = mime.lookup(filename) || opt.defaultContentType || 'application/octet-stream';
+  listStackResources(resources, nextToken) {
```

Comment:
This is copy-paste from my own [serverless-export-env](https://github.com/arabold/serverless-export-env) plugin.

Reply:
As `serverless-s3-deploy` always runs _after_ the first deploy all resources (in specific the S3 bucket of course) should be available in the meantime.

I think we should also add the `Fn::ImportValue` (in a separate PR) as this might be a quite common use case for some users as well.

Reply:
Surely this sort of function / mechanism already exists somewhere in serverless that could be reused?

Reply:
Didn't find it, that's why I implemented it in first place. Seems a bit silly though, I agree.